pkg/codegen/schema.go

```diff
@@ -237,7 +237,7 @@ func PropertiesEqual(a, b Property) bool {
 	return a.JsonFieldName == b.JsonFieldName && a.Schema.TypeDecl() == b.Schema.TypeDecl() && a.Required == b.Required
 }
 
-func GenerateGoSchema(sref *openapi3.SchemaRef, path []string) (Schema, error) {
+func GenerateGoSchema(sref *openapi3.SchemaRef, path []string, component bool) (Schema, error) {
```

Comment:
`component` could do with a better name.

When it's true we are generating a go type for a schema in '#/components/schemas' the root document.